client/asset/btc/btc.go

```diff
@@ -5026,6 +5026,136 @@ func (btc *baseWallet) RefundBond(ctx context.Context, ver uint16, coinID, scrip
 	return NewOutput(txHash, 0, uint64(msgTx.TxOut[0].Value)), nil
 }
 
+func (btc *baseWallet) decodeV0BondTx(msgTx *wire.MsgTx, txHash *chainhash.Hash, coinID []byte) (*asset.BondDetails, error) {
+	if len(msgTx.TxOut) < 2 {
+		return nil, fmt.Errorf("tx %s is not a v0 bond transaction: too few outputs", txHash)
+	}
+	_, lockTime, pkh, err := dexbtc.ExtractBondCommitDataV0(0, msgTx.TxOut[1].PkScript)
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract bond commitment details from output 1 of %s: %v", txHash, err)
+	}
+	// Sanity check.
+	bondScript, err := dexbtc.MakeBondScript(0, lockTime, pkh[:])
+	if err != nil {
+		return nil, fmt.Errorf("failed to build bond output redeem script: %w", err)
+	}
+	pkScript, err := btc.scriptHashScript(bondScript)
+	if err != nil {
+		return nil, fmt.Errorf("error constructing p2sh script: %v", err)
+	}
+	if !bytes.Equal(pkScript, msgTx.TxOut[0].PkScript) {
+		return nil, fmt.Errorf("bond script does not match commit data for %s: %x != %x",
+			txHash, bondScript, msgTx.TxOut[0].PkScript)
+	}
+	return &asset.BondDetails{
+		Bond: &asset.Bond{
+			Version: 0,
+			AssetID: btc.cloneParams.AssetID,
+			Amount:  uint64(msgTx.TxOut[0].Value),
+			CoinID:  coinID,
+			Data:    bondScript,
+			//
+			// SignedTx and UnsignedTx not populated because this is
+			// an already posted bond and these fields are no longer used.
+			// SignedTx, UnsignedTx []byte
+			//
+			// RedeemTx cannot be populated because we do not have
+			// the private key that only core knows. Core will need
+			// the BondPKH to determine what the private key was.
+			// RedeemTx []byte
+		},
+		LockTime: time.Unix(int64(lockTime), 0),
+		CheckPrivKey: func(bondKey *secp256k1.PrivateKey) bool {
+			pk := bondKey.PubKey().SerializeCompressed()
+			pkhB := btcutil.Hash160(pk)
+			return bytes.Equal(pkh[:], pkhB)
+		},
+	}, nil
+}
+
+// FindBond finds the bond with coinID and returns the values used to create it.
+func (btc *baseWallet) FindBond(_ context.Context, coinID []byte, _ time.Time) (bond *asset.BondDetails, err error) {
```

Comment:
And don't declare `*baseWallet`? Then `*baseWallet` will not be a `asset.Bonder`. I may not be understanding, but I think it is better on `*baseWallet` so that it covers any potential new wallets that also cannot fetch blocks.